app-modules/service-management/src/DataTransferObjects/ServiceRequestDataObject.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Aiding App™ is licensed under the Elastic License 2.0. For more details,
+    see <https://github.com/canyongbs/aidingapp/blob/main/LICENSE.>
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AidingApp\ServiceManagement\DataTransferObjects;
+
+use Spatie\LaravelData\Data;
+use Spatie\LaravelData\Attributes\Validation\Nullable;
+
+class ServiceRequestDataObject extends Data
+{
+    public function __construct(
+        public ?string $division_id,
+        public ?string $status_id,
+        public ?string $type_id,
+        public ?string $priority_id,
+        public ?string $title,
+        public string|Nullable $close_details,
+        public string|Nullable $res_details,
+        public ?string $respondent_type,
+        public ?string $respondent_id,
+    ) {}
+
+    /**
+     * Convert the DTO to an array.
+     *
+     * @return array
+     */
+    public function toArray(): array
+    {
+        return [
+            'division_id' => $this->division_id,
+            'status_id' => $this->status_id,
+            'type_id' => $this->type_id,
+            'priority_id' => $this->priority_id,
+            'title' => $this->title,
+            'close_details' => $this->close_details,
+            'res_details' => $this->res_details,
+            'respondent_type' => $this->respondent_type,
+            'respondent_id' => $this->respondent_id,
+        ];
+    }
```

Comment:
Maybe I am wrong, but since we're not doing transformation here do we need this method? Because I think that the base `Data` object has it

And what is the difference between `?string` and `string|Nullable` in this case?